eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -3,93 +3,373 @@
+      return $true
+    }
+    return $false
+}
 
-    if ($purgeableResourcesFromRG) {
-      $purgeableResources += $purgeableResourcesFromRG
-      Write-Verbose "Found $($purgeableResourcesFromRG.Count) potentially purgeable resources in resource group $($rg.ResourceGroupName)"
+function GetDeleteAfterTag([object]$ResourceGroup) {
+  return GetTag $ResourceGroup "DeleteAfter"
+}
+
+function HasExpiredDeleteAfterTag([string]$DeleteAfter) {
+  if ($DeleteAfter) {
+    $deleteDate = $deleteAfter -as [DateTime]
+    return $deleteDate -and [datetime]::UtcNow -gt $deleteDate
+  }
+  return $false
+}
+
+function HasException([object]$ResourceGroup) {
+    if ($Exceptions.Count) {
+      if ($Exceptions.Contains($ResourceGroup.ResourceGroupName)) {
+        Write-Host " Skipping allowed resource group '$($ResourceGroup.ResourceGroupName)' because it is in the allow list '$AllowListPath'"
+        return $true
+      }
+      return $false
+    }
+    if ($ShouldCheckAllowList) {
+      $lines = Get-Content $AllowListPath
+      foreach ($line in $lines) {
+        if ($line -and !$line.StartsWith("#")) {
+          $Exceptions.Add($line.Trim())
+        }
+      }
+      # If allow list is empty or only contains comments, disable allow list checking
+      if (!$Exceptions.Count) {
+        $ShouldCheckAllowList = $false
+        return $false
+      }
+      return HasException $ResourceGroup
+    }
+    return $false
+}
+
+function FindOrCreateDeleteAfterTag {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+  param(
+    [object]$ResourceGroup
+  )
+
+  $deleteAfter = GetTag $ResourceGroup "DeleteAfter"
+  if (!$deleteAfter -or !($deleteAfter -as [datetime])) {
+    $deleteAfter = [datetime]::UtcNow.AddHours($DeleteAfterHours)
+    if ($Force -or $PSCmdlet.ShouldProcess("$($ResourceGroup.ResourceGroupName) [DeleteAfter (UTC): $deleteAfter]", "Update Group DeleteAfter tag")) {
+      Write-Host "Adding DeleteAfer tag with value '$deleteAfter' to group '$($ResourceGroup.ResourceGroupName)'"
+      $ResourceGroup | Update-AzTag -Operation Merge -Tag @{ DeleteAfter = $deleteAfter }
+    }
+  }
+}
+
+function HasDeleteLock() {
+  return $false
+}
+
+function DeleteOrUpdateResourceGroups() {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+  param()
+
+  if ($IsProvisionerApp) {
+    AddGithubUsersToAliasCache
+  }
+
+  Write-Verbose "Fetching groups"
+  $allGroups = @(Get-AzResourceGroup)
+  $toDelete = @()
+  $toUpdate = @()
+  Write-Host "Total Resource Groups: $($allGroups.Count)"
+
+  foreach ($rg in $allGroups) {
+    if (HasException $rg) {
+      continue
+    }
+    $deleteAfter = GetDeleteAfterTag $rg
+    if ($deleteAfter) {
+      if (HasExpiredDeleteAfterTag $deleteAfter) {
+        $toDelete += $rg
+      }
+      continue
+    }
+    if (!$DeleteNonCompliantGroups) {
```

Comment:
Just as an FYI if someone accidently runs this once with this set and then DeleteAfter tags will get added and then our automation will end up cleaning those up, even if we aren't passing this. So we need to be very careful about this testing or running with this option until we are in a clean state.

Reply:
Perhaps we should just remove it as a parameter entirely until we're ready?

Reply:
Are you suggesting that we just comment out this entire section until then? That would probably be the safest thing until you are ready.

Reply:
I was thinking keep the code the same but remove the ability for anyone to set the parameter to true.

Reply:
Or something like this:

```
if ($true -or !$DeleteNonCompliantGroups) {
```

Reply:
Yeah whatever we can do to help prevent accidental deletion.